### DISCUSSION TITLE
UDP and TCP data transfer behavior in Fast DDS large data profile of FASTDDS_BUILTIN_TRANSPORTS

### DISCUSSION THREAD

**Discussion by AnoopGQuest** [eProsima/Fast-DDS#4938] [General]
I'm working with the Fast DDS large data profile and I have a question regarding the observed data transfer behavior during data publication.
According to my understanding from the documentation, the large data profile leverages UDP for participant discovery and TCP for reliable data transfer. However, in my observations, I see the following:

UDP Discovery: As expected, UDP is used for initial discovery messages.
Data Transfer: When I put data size printouts in the UDPSenderResource and TCPSenderResource, I saw data being transferred using both UDP and TCP. Interestingly, the data size seems to be the same on both protocols.

I would appreciate it if someone from the community could clarify the expected behavior in the Fast DDS large data profile builtintransports and explain why I might be seeing both UDP and TCP used for data transfer.

**Comment by MiguelCompany** [comment]
@AnoopGQuest The important bit is the list of destination locators. You would see their kind is UDP only when sending the participant discovery datagrams. In any other case you will see either SHM or TCP.
